Improve handling of invisible sequences in trim mode

stroncium opened the pull request:
Added tests from #32 and fixed code to pass them.

sindresorhus commented:
I merged https://github.com/chalk/wrap-ansi/pull/32, can you rebase and remove the `.failing` part from that test instead? I think it makes more sense to keep it as a separate test.

sindresorhus commented:
// @coreyfarrell

coreyfarrell commented:
@sindresorhus I've just realized another question - should strings be wrapped before or after trimming?  Currently wrapping is done before trimming which produces the following results:

```js
// 3 spaces before / between / after
t.is(wrapAnsi('   foo   bar   ', 6), 'foo\nbar');
t.is(wrapAnsi('   foo   bar   ', 3), '\nfoo\nbar\n');
```

stroncium commented:
@coreyfarrell @sindresorhus It's not a question of trim first or not(trimming is not just a stage), it's a bug, and I guess part of it was introduced by my big patch. Anyhow, added test, fixed.

coreyfarrell commented:
@stroncium I just verified that all `cliui` tests now pass when `wrap-ansi` code is updated with this PR.  Thanks!

coreyfarrell commented on index.js:
```diff
@@ -60,6 +60,30 @@ const wrapWord = (rows, word, columns) => {
+			break;
+		}
+
+		first++;
```
The tests never hit this line.  Is it possible for `words.length > 0 && stringWidth(words[0]) === 0`?

stroncium replied:
Yeah, in actual code left side is always trimmed already. Changed to trimRight

coreyfarrell commented on index.js:
```diff
@@ -60,6 +60,23 @@ const wrapWord = (rows, word, columns) => {
+		last--;
+	}
+
+	return words.slice(0, first).join('') + words.slice(first, last).join(' ') + words.slice(last, words.length).join('');
```
Please remove `words.slice(0, first).join('')` as it is always an empty string.

Personally I'd also remove the `first` variable entirely and just use `0` where appropriate though I don't feel strongly about this.

coreyfarrell replied:
@stroncium ping

@sindresorhus do you see any other issues with this PR?

stroncium replied:
@coreyfarrell done with a bonus. Sorry for taking so long. I suppose github decided I've read this when I glanced over email. =/ (Particularly annoying feature, if anyone knows how to disable this I'd be happy to listen.)

coreyfarrell replied:
Github includes a link to a 1x1 pixel image like `<img src="https://github.com/beacon/id...">`.  Maybe if you've told your email client that github emails are trusted this bypasses the normal privacy protection?  Every github email shows me the warning `To protect your privacy, Thunderbird has blocked remote content in this message.`

stroncium replied:
Well, we'll see. I'm using Inbox, but it gets shut down today, hopefully gmail got this feature.

sindresorhus commented:
Thanks for fixing, @stroncium 👍
